engine/src/main/java/org/destinationsol/game/SolGame.java

```diff
@@ -193,7 +189,7 @@ private void createPlayer(String shipName) {
         String itemsStr = !respawnItems.isEmpty() ? "" : shipConfig.items;
 
         boolean giveAmmo = shipName != null && respawnItems.isEmpty();
-        hero = shipBuilder.buildNewFar(this, new Vector2(pos), null, 0, 0, pilot, itemsStr, hull, null, true, money, new TradeConfig(), giveAmmo).toObj(this);
+        hero = new Hero(shipBuilder.buildNewFar(this, new Vector2(pos), null, 0, 0, pilot, itemsStr, hull, null, true, money, new TradeConfig(), giveAmmo).toObj(this));
```

Comment:
Inherited issue: we should probably do something about the excessive arguments passed to the `buildNewFar` method. It does not look like `ShipBuilder` follows the Builder pattern to me.

Reply:
Yes, DestSol has way too many arguments in every single method. That being said, I kinda don't like the builder pattern for this particular case, since almost nothing is optional, and one of the biggest advantages of builder pattern is the ability to leave something out, so that the default value can be used for that parameter. Rather than have mandatory calls to 10 different methods, I would rather have them in the constructor.

Reply:
Assumed that the class followed the builder pattern because it has the prefix `Builder` :) Are those 10 arguments really unavoidable? I see a couple of nulls and 0s there. Can they be omitted?

Reply:
I haven't checked every single argument for ShipBuilder, I gave a general statement for most of the classes. I'll have a second look at ShipBuilder arguments though.